packages/platform/types/bookings/2024-08-13/inputs/create-booking.input.ts

```diff
@@ -1,24 +1,51 @@
 import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
 import { Type } from "class-transformer";
+import type { ValidationArguments, ValidationOptions } from "class-validator";
 import {
   IsInt,
   IsDateString,
   IsTimeZone,
   IsEnum,
-  IsEmail,
   ValidateNested,
   IsArray,
   IsString,
+  isEmail,
   IsOptional,
   IsUrl,
   IsObject,
   IsBoolean,
   Min,
+  registerDecorator,
+  Validate,
 } from "class-validator";
+import { isValidPhoneNumber } from "libphonenumber-js";
 
 import type { BookingLanguageType } from "./language";
 import { BookingLanguage } from "./language";
 
+function RequireEmailOrPhone(validationOptions?: ValidationOptions) {
```

Comment:
We should also add e2e tests for booking with phone in `apps/api/v2/src/ee/bookings/2024-08-13/controllers/user-bookings.controller.e2e-spec.ts` or can copy test setup from the file and create a new `phone-bookings.e2e-spec.ts` file.